src/Opdex.Platform.Application/EntryHandlers/Tokens/CreateAddTokenCommandHandler.cs

```diff
@@ -19,42 +24,79 @@ public class CreateAddTokenCommandHandler : IRequestHandler<CreateAddTokenComman
 {
     private readonly IMapper _mapper;
     private readonly IMediator _mediator;
+    private readonly InterfluxConfiguration _interfluxConfiguration;
+    private readonly ILogger<CreateAddTokenCommandHandler> _logger;
 
-    public CreateAddTokenCommandHandler(IMapper mapper, IMediator mediator)
+    public CreateAddTokenCommandHandler(IMapper mapper, IMediator mediator, InterfluxConfiguration interfluxConfiguration,
+        ILogger<CreateAddTokenCommandHandler> logger)
     {
         _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
         _mediator = mediator ?? throw new ArgumentNullException(nameof(mediator));
+        _interfluxConfiguration = interfluxConfiguration ?? throw new ArgumentNullException(nameof(interfluxConfiguration));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
     }
 
     public async Task<TokenDto> Handle(CreateAddTokenCommand request, CancellationToken cancellationToken)
     {
-        var token = await _mediator.Send(new RetrieveTokenByAddressQuery(request.Token, findOrThrow: false));
-        if (!(token is null))
+        var token = await _mediator.Send(new RetrieveTokenByAddressQuery(request.Token, findOrThrow: false), CancellationToken.None);
+        if (token is not null)
         {
             throw new TokenAlreadyIndexedException(request.Token);
         }
 
-        var latestBlock = await _mediator.Send(new GetBestBlockReceiptQuery());
+        var latestBlock = await _mediator.Send(new GetBestBlockReceiptQuery(), CancellationToken.None);
 
         StandardTokenContractSummary summary;
         try
         {
             summary = await _mediator.Send(new CallCirrusGetStandardTokenContractSummaryQuery(request.Token,
                                                                                               latestBlock.Height,
                                                                                               includeBaseProperties: true,
-                                                                                              includeTotalSupply: true));
+                                                                                              includeTotalSupply: true), CancellationToken.None);
         }
         catch (Exception)
         {
             throw new InvalidDataException("token", "Unable to validate SRC token.");
         }
 
+        var interfluxSummary = await _mediator.Send(new CallCirrusGetInterfluxTokenContractSummaryQuery(request.Token, latestBlock.Height), CancellationToken.None);
+        if (interfluxSummary is not null && !ValidateWrappedToken(interfluxSummary))
+        {
+            throw new InvalidDataException("token", "Wrapped token must be owned by Interflux multisig wallet.");
```

Comment:
This feels the best thing to do, though feeling conflicted about it. Any token which is created via `CreateTokenCommandHandler` will still be persisted, just without the `TokenChain` interflux mapping.